src/webgpu/api/validation/capability_checks/limits/limit_utils.ts

```diff
@@ -1005,6 +1005,29 @@ export class LimitTestsImpl extends GPUTestBase {
     await this.expectValidationError(test, shouldError, msg);
   }
 
+  /**
+   * Test a method on GPURenderCommandsMixin or GPUBindingCommandsMixin
+   * The function will be called with the mixin.
+   */
+  async testGPURenderAndBindingCommandsMixin(
+    encoderType: RenderEncoderType,
+    fn: ({
+      mixin,
+      bindGroup,
+    }: {
+      mixin: GPURenderCommandsMixin & GPUBindingCommandsMixin;
```

Comment:
All 2 interfaces that implement GPURenderCommandsMixin also implement GPUBindingCommandsMixin. I think this and `testGPURenderCommandsMixin` can just be a single function:

```ts
  async testGPURenderCommandsMixin(
    encoderType: RenderEncoderType,
    fn: (args: {
      mixin: GPURenderCommandsMixin & GPUBindingCommandsMixin;
      bindGroup: GPUBindGroup;
    }) => void,
    shouldError: boolean,
    msg = ''
  ) {
    const { mixin, prep, test, bindGroup } = this._getGPURenderCommandsMixin(encoderType);
    fn({ mixin, bindGroup });
    prep();

    await this.expectValidationError(test, shouldError, msg);
  }
```

Reply:
done